src/components/Modal.js

```diff
@@ -10,26 +10,33 @@ import { LabeledButton } from './buttons';
 
 /**
  * @typedef ModalBaseProps
- * @prop {() => void} onCancel - `onCancel` is required for Modals
- */
-
-/**
- * @typedef {DialogProps & ModalBaseProps} ModalProps
+ * @prop {boolean} [closeOnExternalInteraction=true] - By default, a modal
+ *   with a provided `onCancel` callback will close when a user clicks outside
+ *   the modal, or focus is moved outside of the modal. In some cases, this
+ *   may not be desireable. When this option is disabled, the modal may still
+ *   be closed by pressing ESC.
```

Comment:
Hmm. How strong a reason do we have to continue supporting closing the dialog when clicking outside of it? IIRC the original reason for this was to avoid needing to do focus trapping (ie. preventing focus from moving outside of the dialog). If we are going to prevent dialogs closing by interacting with external elements then I think we'll need to implement focus trapping as well.

Reply:
Ooh! Thank you for thinking this through and not letting me proceed blindly.

Let's see... the close behavior as initially implemented in this hook was intended for things like menus which _should_ go away if the user clicks or focuses away, but for modals, this behavior as default (for modals/dialogs) may not make sense.

I just did a quick sweep of all uses of `<Modal>`, `<ConfirmModal>` and `<Dialog>` in `client` and `lms` and am of the opinion that close-on-external-click-or-focus should _not_ be the default behavior. (ESC should still close, however).

Unless you demur, I'm going to go ahead and make this change.